### PR TITLE
Minor updates to opacity_transfer_function

### PR DESCRIPTION
**Pull request by banesullivan**
I needed to control the `kind` of interpolation for some figures I am making with opacities and figured I'd update the docs for this method.

Here is a snippet from a bigger project:

```py
opacity = [0.2, 0.2, 0.75, 0.2, 0.2]
# 'linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic', 'previous', 'next'
tf = pv.opacity_transfer_function(opacity, 256, kind='slinear')
vv =np.linspace(*dargs_vel["clim"], len(tf))

plt.plot(vv, tf.astype(float) / 255.)
plt.axvline(5500, color='r', linestyle='--')
plt.ylim(0,1)
plt.title('My Opacity Transfer Function')
plt.ylabel('Opacity')
plt.xlabel('Scalar Values')
plt.show()
```

download

and then pass the transfer function as `opacity` argument to `add_mesh`
```py
...
p.add_mesh(s2_refrac.threshold(), opacity=tf, **dargs_vel)
p.add_mesh(s2_refrac.contour([5500]), line_width=5, 
           opacity=tf, **dargs_vel)
...
```

Screen Shot 2020-02-05 at 12 19 38 PM

### CODE REVIEW

**Comment by banesullivan** [comment]
One day, it would be super useful to implement a few sliders to control the opacity transfer function much like @marcomusy's awesome work in `vtkplotter`:

58336107-5a09a180-7e43-11e9-8c4e-b50e4e95ae71

File: pyvista/plotting/tools.py [hunk 1]
```diff
@@ -180,8 +180,57 @@ def normalize(x, minimum=None, maximum=None):
...
-    """Get the opacity transfer function results: range from 0 to 255."""
+def opacity_transfer_function(mapping, n_colors, interpolate=True,
+                              kind='quadratic'):
+    """Get the opacity transfer function for a mapping.
+
+    These values will map on to a scalar bar range and thus the number of
+    colors (``n_colors``) must correspond to the number of colors in the color
+    mapping that these opacities are associated to.
+
+    If interpolatig, ``scipy.interpolate.interp1d`` is used if available,
```
Comment (akaszynski) [inline pyvista/plotting/tools.py]: "interpolatig" seems like a spelling error.

Comment (banesullivan) [reply pyvista/plotting/tools.py]: Hah thanks, ran codespell and fixed a lot of other typos as well

**Comment by GuillaumeFavelier** [comment]
I investigated the spelling issues and `codespell` should be able to detect those @banesullivan, I opened #591 to fix this

**Pull request merged**